packages/zosfiles/src/methods/copy/Copy.ts

```diff
@@ -201,13 +226,42 @@ export class Copy {
             const downloadDir = path.join(tmpdir(), fromPds);
             await Download.allMembers(session, fromPds, {directory:downloadDir});
             const uploadFileList: string[] = ZosFilesUtils.getFileListFromPath(downloadDir);
+            const truncatedMembers: string[] = [];
 
             for (const file of uploadFileList) {
-                const uploadingDsn = `${toPds}(${ZosFilesUtils.generateMemberName(file)})`;
-                const uploadStream = IO.createReadStream(file);
-                await Upload.streamToDataSet(session, uploadStream, uploadingDsn);
+                const memName = ZosFilesUtils.generateMemberName(file);
+                const uploadingDsn = `${toPds}(${memName})`;
+                try {
+                    const uploadStream = IO.createReadStream(file);
+                    await Upload.streamToDataSet(session, uploadStream, uploadingDsn);
+                }
+                catch(error) {
+                    if(error.message && error.message.includes("Truncation of a record occurred during an I/O operation")) {
+                        truncatedMembers.push(memName);
+                    }
+                    else {
+                        Logger.getAppLogger().error(error);
+                    }
+                    continue;
+                }
+            }
+            const truncatedMembersFile = path.join(tmpdir(), 'truncatedMembers.txt');
+            if(truncatedMembers.length > 0) {
+                const firstTenMembers = truncatedMembers.slice(0, 10);
+                fs.writeFileSync(truncatedMembersFile, truncatedMembers.join('\n'));
+                const numMembers = truncatedMembers.length - firstTenMembers.length;
+                return {
+                    success: true,
+                    commandResponse:
+                        ZosFilesMessages.datasetCopiedSuccessfully.message + " " +
+                        ZosFilesMessages.membersContentTruncated.message + "\n\n" +
+                        firstTenMembers.join('\n') +
+                        `\n... and ${numMembers} more` +
```

Comment:
Could this logic result in the message "and 0 more" ?

Reply:
This is definitely a judgement call. If I were to hit the edge-case of exactly 10, I would think it odd to be told that "0 more" occurred. Alternatively if someone were automating on the output, it might be useful to know that "x more" would always be part of the message. I lean to removing the message when it would display "0 more". Perhaps wait a bit to see if others offer opinions.

Reply:
i agree that adding "...and X more" should only be included if X is greater than 1